src/client/js/services/RevisionCompareContainer.js

```diff
@@ -0,0 +1,152 @@
+import { Container } from 'unstated';
+import loggerFactory from '@alias/logger';
+
+import { toastError } from '../util/apiNotification';
+
+const logger = loggerFactory('growi:PageHistoryContainer');
+
+/**
+ * Service container for personal settings page (RevisionCompare.jsx)
+ * @extends {Container} unstated Container
+ */
+export default class RevisionCompareContainer extends Container {
+
+  constructor(appContainer, pageContainer) {
+    super();
+
+    this.appContainer = appContainer;
+    this.pageContainer = pageContainer;
+
+    this.state = {
+      errMessage: null,
+
+      fromRevision: null,
+      toRevision: null,
+
+      revisions: [],
+    };
+
+    this.readyRevisions = this.readyRevisions.bind(this);
+    this.fetchPageRevisionBody = this.fetchPageRevisionBody.bind(this);
+    this.fetchAllPageRevisions = this.fetchAllPageRevisions.bind(this);
+    this.fetchPageRevision = this.fetchPageRevision.bind(this);
+    this.handleFromRevisionChange = this.handleFromRevisionChange.bind(this);
+    this.handleToRevisionChange = this.handleToRevisionChange.bind(this);
+  }
+
+  /**
+   * Workaround for the mangling in production build to break constructor.name
+   */
+  static getClassName() {
+    return 'RevisionCompareContainer';
+  }
+
+  async readyRevisions() {
+    await this.fetchAllPageRevisions();
+
+    const latestRevisionId = this.state.revisions[0]._id;
+    const { compareRevisionIds } = this.pageContainer.state;
+    const fromRevisionIdParam = compareRevisionIds[0] || latestRevisionId;
+    const toRevisionIdParam = compareRevisionIds[1] || latestRevisionId;
+    await this.handleFromRevisionChange(fromRevisionIdParam);
+    await this.handleToRevisionChange(toRevisionIdParam);
+  }
+
+  /**
+   * Fetch page revision body by revision_id in argument
+   * @param {string} revisionId
+   */
+  async fetchPageRevisionBody(revisionId) {
+    const { pageId, shareLinkId } = this.pageContainer.state;
+    try {
+      const res = await this.appContainer.apiv3Get(`/revisions/${revisionId}`, { pageId, shareLinkId });
+      if (res == null || res.data === undefined || res.data.revision === undefined) {
+        logger.warn(`cannot get revision. revisionId: ${revisionId}`);
+        return null;
+      }
+      return res.data.revision.body;
+    }
+    catch (err) {
+      toastError(err);
+      this.setState({ errorMessage: err.message });
+      logger.error(err);
+    }
+  }
+
+  /**
+   * Fetch all page revisions
+   *
+   * Each revision to be saved contains only "_id" and "createdAt", and "body" is initialized to null.
+   * ex. [{_id: "5ff03fded799ebc858a09266", body: null, creat…}, {_id: "5ff03fbed799ebc858a09262", body: null, creat…}]
+   */
+  async fetchAllPageRevisions() {
```

Comment:
I think it is not good situation that needs fetching all revisions.
Please consider another UI that need not to fetch all.